src/isolation/tempdir.rs

```diff
@@ -1,25 +1,114 @@
-use std::{fs::Permissions, os::unix::fs::PermissionsExt, path::PathBuf};
+use std::{
+	fs::{DirBuilder, Permissions},
+	os::unix::fs::{DirBuilderExt, PermissionsExt},
+	path::PathBuf,
+};
 
 use tempfile::{Builder, TempDir, env};
 use uuid::Uuid;
 
 /// Creates a temporary directory.
 ///
+/// To keep the temporary directory clean and because we need to include the
+/// parent directory in [`crate::isolation::landlock::initialize`], a sub-folder
+/// called `uhyve-{uid}` (e.g. `/tmp/uhyve-1000`) is used to create temporary
+/// directories under.
```

Comment:
nitpick: `under.` feels like the sentence is incomplete, perhaps put something like `there` after `under`, or just replace `directories under` with `further subdirectories`.

Reply:
*under (the aforemementioned subfolder).

I interpreted the "sub-folder" as a "location" or a state.